contracts/token/HolographUtilityToken.sol

```diff
@@ -126,9 +126,21 @@ contract HolographUtilityToken is ERC20H {
    * @param initPayload abi encoded payload to use for contract initilaization
    */
   function init(bytes memory initPayload) external override returns (bytes4) {
-    address contractOwner = abi.decode(initPayload, (address));
+    (address contractOwner, uint256 tokenAmount, uint256 targetChain, address tokenRecipient) = abi.decode(
+      initPayload,
+      (address, uint256, uint256, address)
+    );
     _setOwner(contractOwner);
-    HolographERC20Interface(msg.sender).sourceMint(contractOwner, 10000000 * (10**18));
+    /*
+     * @dev Mint token only if target chain matches current chain.
+     *      Goal of this is to restring minting on Ethereum only for mainnet deployment.
```

Comment:
is restring supposed to be `restrict`?

Also how is this enforced if the target chain is passed in via the payload. would it be better to hardcode the target chain to ethereum?

Reply:
yep it's a typo

Reply:
fixed in next commit